src/tests/sqlsmith/src/relation.rs

```diff
@@ -32,28 +32,30 @@ fn create_join_on_clause(left: String, right: String) -> Expr {
 
 impl<'a, R: Rng> SqlGenerator<'a, R> {
     /// A relation specified in the FROM clause.
-    pub(crate) fn gen_from_relation(&mut self) -> TableWithJoins {
-        match self.rng.gen_range(0..=9) {
-            0..=7 => self.gen_simple_table(),
-            8..=8 => self.gen_time_window_func(),
+    pub(crate) fn gen_from_relation(&mut self) -> (TableWithJoins, Vec<Table>) {
+        match self.rng.gen_range(0..=50) {
+            0..=39 => self.gen_simple_table(),
+            40..=44 => self.gen_time_window_func(),
             // TODO: Enable after resolving: <https://github.com/singularity-data/risingwave/issues/2771>.
-            9..=9 => self.gen_equijoin_clause(),
-            // TODO: Currently `gen_subquery` will cause panic due to some wrong assertions.
-            10..=10 => self.gen_subquery(),
+            45..=49 => self.gen_equijoin_clause(),
+            50..=50 => self.gen_table_subquery(),
```

Comment:
We should decouple recursion and weights, `gen_table_subquery` has higher weightage of being executed, but this chance decreases with depth of query.

This can be done by wrapping body of `gen_table_subquery` in `self.can_recurse()` and returning `gen_simple_table`, if `self.can_recurse()` doesn't hold.

Reply:
Let probability of table subquery be x 

isn,t this just add the chance of simple table by x/2
and reduce table subquery by x/2.

It doesn,t reduce with depth of query. 

One way I can think about it is to pass it thru global/local parameter. 
if I  use global will add complexity
If I use local parameter will have many refactor, 

That why I just reduce the probability to very low.

Let me think about it see if any solution.

Reply:
> isn,t this just add the chance of a simple table by x0.7
and reduce table subquery by x0.7.

Using `self.can_recurse` makes it easier to adjust chance of recursion. This applies to all generators which can recurse. e.g. `gen_with, gen_subquery`. If we use the weight to indicate this, we have to adjust each individual weights instead for each generator instead.

Reply:
I think I will use `self.recurse` to set the range in `gen_from_relation` instead. I can,t see a way in `gen_table_subquery`.

Reply:
> I think I will use self.recurse to set the range in gen_from_relation instead. I can,t see a way in gen_table_subquery.

That works better.

Reply:
Now I set gen_table_subquery with 1/11 chance. with self.can_recursive.
it takes about 53s - 64s to compile a test, and in total 249.321s. is it acceptable?
Besides, I think we really need to reduce chance of recursion when the depth getting higher, the current one, I dun think the chance decreased gradually because it is not accumulated

Reply:
> Besides, I think we really need to reduce chance of recursion when the depth getting higher, the current one, I dun think the chance decreased gradually because it is not accumulated (Not in this PR)

Let's keep sqlsmith simple for now. But agreed, in the future we can design a better way to control recursion.